Lib/subprocess.py

```diff
@@ -908,6 +913,10 @@ def __exit__(self, exc_type, value, traceback):
             self.wait()
 
     def __del__(self, _maxsize=sys.maxsize, _warn=warnings.warn):
+        if _mswindows:
+            # We don't need to do anything in Windows. See `_cleanup` comment
+            # in `Popen.__init__` for more info.
+            return
```

Comment:
That's wrong: you must also wait for process completion on Windows.

Reply:
Hi @vstinner ! Thanks for the review! On Windows we just need to close the handle when we no longer need it, so that the kernel can free it. CloseHandle is called as a `__del__` in `class Handle` defined above, so there are no additional actions required. From [1]

```
When you no longer need these handles, close them by using the CloseHandle function.
```

[1] https://docs.microsoft.com/en-us/windows/desktop/ProcThread/creating-processes

Reply:
@vstinner If we're in `__del__`, presumably that means that nobody has a reference to the `Popen` object, right? In that case, we can let it be cleaned up. Closing the handle does not terminate the process - it'll just keep running and we can't get back to it through that particular handle (but we could use `OpenProcess` with the PID to get a new handle later).